src/core/amount.js

```diff
@@ -626,22 +598,26 @@ function(quality, counterCurrency, counterIssuer, opts) {
     if (this._is_native) {
       // pay:$price              drops  get:1 X
       // pay:($price / 1,000,000)  XRP  get:1 X
-      nativeAdjusted = adjusted.div(Amount.bi_xns_unit);
+      nativeAdjusted = nativeAdjusted.divide(bi_xns_unit);
     } else if (baseCurrency.is_valid() && baseCurrency.is_native()) {
       // pay:$price X                   get:1 drop
       // pay:($price * 1,000,000) X     get:1 XRP
-      nativeAdjusted = adjusted.times(Amount.bi_xns_unit);
+      nativeAdjusted = nativeAdjusted.multiply(bi_xns_unit);
     }
   }
+  if (this._is_native) {
+    this._set_value(
+      new XRPValue(nativeAdjusted.round(6, Value.getBNRoundDown())));
```

Comment:
It looks like this is passing an IOUValue instance to the constructor of XRPValue.